reactive/kotlinx-coroutines-rx2/test/SchedulerStressTest.kt

```diff
@@ -0,0 +1,115 @@
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+        testRunnableDisposed(scheduler::scheduleDirect)
+    }
+
+    @Test
+    fun testSchedulerWorkerDisposed(): Unit = runTest {
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+        testRunnableDisposed(scheduler.createWorker()::schedule)
+    }
+
+    private suspend fun testRunnableDisposed(block: RxSchedulerBlockNoDelay) {
+        expect(1)
+
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+
+        val n = 2000 * stressTestMultiplier
+        coroutineScope {
+            repeat(n) { i ->
+                launch {
+                    val a = ByteArray(1000000) //1MB
+                    val disposable = block(Runnable {
+                        expectUnreached()
+                        runBlocking {
+                            keepMe(a)
+                        }
+                    })
+                    disposable.dispose()
+                    expect(i + 2)
+                }
+                yield()
+            }
+        }
+
+        scheduler.shutdown()
+        finish(n + 2)
+    }
+
+    /**
+     * Test function that holds a reference. Used for testing OOM situations
+     */
+    private suspend fun keepMe(a: ByteArray) {
+        delay(10)
+    }
+
+    /**
+     * Test that we don't get an OOM if we schedule many delayed jobs at once. It's expected that if you don't dispose that you'd
+     * see a OOM error.
+     */
+    @Test
+    fun testSchedulerDisposedDuringDelay(): Unit = runBlockingTest {
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+        testRunnableDisposedDuringDelay(scheduler::scheduleDirect)
+    }
+
+    @Test
+    fun testSchedulerWorkerDisposedDuringDelay(): Unit = runBlockingTest {
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+        testRunnableDisposedDuringDelay(scheduler.createWorker()::schedule)
+    }
+
+    private suspend fun TestCoroutineScope.testRunnableDisposedDuringDelay(block: RxSchedulerBlockWithDelay) {
+        expect(1)
+
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+
+        val n = 2000 * stressTestMultiplier
+        coroutineScope {
+            repeat(n) { i ->
+                val a = ByteArray(1000000) //1MB
+                val delayMillis: Long = 10
+                val disposable = block(Runnable {
+                    runBlocking {
+                        keepMe(a)
+                    }
+                }, delayMillis, TimeUnit.MILLISECONDS)
+                disposable.dispose()
+                advanceTimeBy(delayMillis)
+                expect(i + 2)
```

Comment:
Again, this `expect` isn't doing anything. It's not running concurrently with any other coroutines, so there's no ambiguous ordering to verify. It would be testing something if it were inside the `Runnable` above.